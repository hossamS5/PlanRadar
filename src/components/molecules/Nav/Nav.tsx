import React from "react";
import Container from "../../atoms/Container/Container";
import { PlanradarLogo, UsersIcon } from "../../../assets/icons";

const Nav: React.FC = () => {
  return (
    <header className="h-[96px] inset-x-0 top-0 z-50  text-mainTheme sticky bg-white shadow-md">
      <Container classess="h-full">
        <nav className="flex items-center gap-16 justify-between w-full h-full">
          <PlanradarLogo />

          <div className="flex items-center gap-2">
            <span className="text-primary-700 text-sm font-medium">
              Hossam Salem
            </span>
            <UsersIcon />
          </div>
        </nav>
      </Container>
    </header>
  );
};

export default Nav;
