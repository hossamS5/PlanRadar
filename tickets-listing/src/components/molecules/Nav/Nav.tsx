import React from "react";
import Container from "../../atoms/Container/Container";
import { PlanradarLogo, UsersIcon } from "../../../assets/icons";

const Nav: React.FC = () => {
  return (
    <header className="h-[96px] inset-x-0 top-0 z-50  text-mainTheme sticky bg-white">
      <Container classess="h-full">
        <nav className="flex items-center gap-16 justify-between w-full h-full">
          <PlanradarLogo />

          <UsersIcon />
        </nav>
      </Container>
    </header>
  );
};

export default Nav;
