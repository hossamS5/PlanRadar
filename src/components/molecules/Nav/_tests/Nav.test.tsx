import { render } from "@testing-library/react";
import Nav from "../Nav";
import { vi } from "vitest";

vi.mock("../../../../assets/icons", () => ({
  PlanradarLogo: () => <svg data-testid="PlanradarLogo" />,
  UsersIcon: () => <svg data-testid="UsersIcon" />,
}));

describe("Nav", () => {
  it("match snapshot", () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });
});
