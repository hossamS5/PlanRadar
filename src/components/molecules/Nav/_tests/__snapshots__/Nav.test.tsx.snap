// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > match snapshot 1`] = `
<DocumentFragment>
  <header
    class="h-[96px] inset-x-0 top-0 z-50  text-mainTheme sticky bg-white shadow-md"
  >
    <div
      class="appContainer w-full h-full"
    >
      <nav
        class="flex items-center justify-between w-full h-full gap-16"
      >
        <svg
          data-testid="PlanradarLogo"
        />
        <div
          class="flex items-center gap-2"
        >
          <span
            class="text-sm font-medium text-center text-primary-700"
          >
            Hossam Salem
          </span>
          <svg
            data-testid="UsersIcon"
          />
        </div>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
